cocotb/handle.py

```diff
@@ -438,10 +451,19 @@ def __hash__(self):
         return SimHandleBase.__hash__(self)
 
 class ConstantObject(NonHierarchyObject):
-    """Constant objects have a value that can be read, but not set.
+    """A constant object.
+
+    Constant objects have a value that can be read, but not set.
```

Comment:
Perhaps
```
An object which has a value that can be read but not set
```
which then both achieves your goal of being a singular noun, but also provides more information than the class name

Reply:
Thanks, done.